File: 第2章 几何绘图/2.4 绕几何中心旋转/绕几何中心旋转画图.py
'''
程序：绕几何中心旋转图形
作者：苏秦@小海豚科学馆公众号
来源：图书《趣味编程三剑客：从Scratch迈向Python和C++》
'''
from turtle import *

def draw_polygon(c):
    '''画正多边形'''
    up()
    fd(c/4/2); left(90); fd(c/4/2)
    down()
    for i in range(4):
        left(90)
        fd(c/4)
    up()
    fd(0 - c/4/2); left(90); fd(c/4/2)
        
def rotate_figure(c, t):
    '''绕几何中心旋转图形'''
    for i in range(t):
        draw_polygon(c)
        left(90/t)

def main():
    '''主程序'''
    mode('logo')
    pencolor('black')
    pensize(4)
    seth(90)
    c = 500             # 周长
    t = 2               # 个数
    rotate_figure(c, t) # 旋转画图

if __name__ == '__main__':
    main()

File: 第2章 几何绘图/2.4 绕几何中心旋转/绕几何中心旋转画图.cpp
/*
程序：绕几何中心旋转图形 
作者：苏秦@小海豚科学馆公众号
来源：图书《趣味编程三剑客：从Scratch迈向Python和C++》
*/
// 画正多边形
void draw_polygon(int c)
{
	pen.up();
	pen.fd(c/4/2); pen.lt(90); pen.fd(c/4/2);
	pen.down();
	for (int i = 0; i < 4; i++) {
		pen.lt(90);
		pen.fd(c/4);
	}
	pen.up();
	pen.fd(0 - c/4/2); pen.lt(90); pen.fd(c/4/2);
}

// 绕几何中心旋转图形
void rotate_figure(int c, int t)
{
    for (int i = 0; i < t; i++) {
        draw_polygon(c);
        pen.lt(90.0/t);
	}
}

// 主程序
int main()
{
	pen.color(_black);
	pen.size(4);
	pen.angle(90);
    int c = 500;         // 周长
	int t = 2;           // 个数
    rotate_figure(c, t); // 旋转画图
	return 0;
}